### DISCUSSION TITLE
Crazyflie with ELRS or 4-in-1

### DISCUSSION THREAD

**Discussion by bzz1337** [bitcraze/discussions#1103] [Q&A]
Hi all, I am trying to figure out how to fly the Crazyflie 2.1 drone with ELRS or Radiomaster's 4-in-1 remote, but didn't find much info about it. Could you please point me where I can start to get this working?

**Comment by knmcguire** [comment]
Ah this is perhaps a bit out of my ballpark and a better question for @tobbeanton or @matejkarasek. But let me at least try to point you in the right direction in terms of info:

Blogposts:
- https://www.bitcraze.io/2022/11/fpv-deck-hack/
- https://www.bitcraze.io/2020/01/crazyflie-bolt-fpv-meets-autonomy/

Code:
- https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/extrx.c

Also check out the doc for the bolt and the bigquad deck:
- https://www.bitcraze.io/products/bigquad-deck/
- https://www.bitcraze.io/products/crazyflie-bolt-1-1/

Hope that this will help you further. If you stil have questions, let us know! Tobias should be back next week so he will be able to help you out with any remaining questions.

**Comment by matejkarasek** [comment]
Hi,
If you have a 4-in-1 transmitter module in your Radiomaster, you just need to flash it with a modified [DIY-Multiprotocol-TX-Module](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module) firmware which includes the "cflie" protocol.
I gave more details on how to do this in #984, please let me know if you need further help!

**Reply by matejkarasek** [reply]
Otherwise, to use ELRS you would need to add an external receiver to your Crazyflie.

If you can find one that supports (C)PPM (single wire)interface this is rather easy, e.g. with the BigQuad deck or even directly with a small hack. 
S-bus interface would go to RX&TX pins but this would require a new driver on the Crazyflie side.
